### DISCUSSION TITLE
Question regarding IOI telemetry

### DISCUSSION THREAD

**Discussion by EndEmpire** [thepeacockproject/Peacock#301] [Q&A] [answered]
Hello Peacock Project team!
    Allow me to start by saying that I greatly appreciate your work to both provide a smooth experience to users and preserve Hitman games for future generations. I am however curious if your project protects against IOI's telemetry as I could not find anything in your projects regarding telemetry (though I may have simply missed it).

I read through some of IOI's privacy policy and it does not inspire confidence, at least for me. So I would be glad to know if I can play Hitman without having to worry about telemetry by using your project. Thank you for your time.

Edit: Just to be clear, I understand if protecting against IOI's telemetry is beyond the scope of this project. I just thought I would ask just in case. Thank you once more.

Section on data collection from [IOI's privacy policy](https://ioi.dk/legal/privacy-policy):

WHAT INFORMATION WE COLLECT

2.1 WHEN YOU CONNECT TO OUR GAMES

Whenever you connect to an online game, certain information will be collected. The same is true for our games, and for this purpose we collect information to establish and maintain your connection to our games. This information includes:

Connection Information. When you connect to our games, our servers automatically collect some information. This information includes your Internet Protocol address, your chosen platform with which you access this game, the date and time of your request and information about your ownership of this game.

Our legal basis for the processing of this information is GDPR Art. 6 (1) (b).

Device Information. We may collect information about the device you are using to access our websites or play our games, including what type of device it is, what operating system you are using, your device settings, unique device identifiers and crash data.

Whether we collect some or all this information depends on what type of device you are using and its settings. Please check the policies of your device manufacturer or software provider to learn more about what information your device makes available to us.

Our legal basis for the processing of this information is GDPR Art. 6 (1) (b).

**Comment by grappigegovert** [comment, marked as answer]
While connected to a Peacock instance, all data, including the telemetry, is sent to Peacock. Peacock currently does nothing with this data.

It is possible for plugins to hook into this data, but I don't think any plugin has been released that does.

The relevant code that handles the incoming metrics is here:
https://github.com/thepeacockproject/Peacock/blob/4e70d55deec3926934a926fc3cee90d4d87ff4ab/components/index.ts#L240-L250

**Reply by EndEmpire** [reply]
Thank you very much! Glad to hear that I can play in peace!

**Comment by LennardF1989** [comment]
We obviously welcome you to play with Peacock (pun intended)! But if you would like to enjoy the real version, but are concerned about the metrics, you should be able to block traffic to metrics.hitman.io using your [hosts file](https://www.howtogeek.com/27350/beginner-geek-how-to-edit-your-hosts-file/). All they know about you after that is what you perform in-game to calculate your score. This is basically what the metrics do too, by the way, except they also keep track of things you click in the menu.

**Reply by EndEmpire** [reply]
This is good to know, thank you very much.
